lib/ansible/modules/storage/netapp/na_cdot_volume.py

```diff
@@ -0,0 +1,432 @@
+# MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+# GNU General Public License for more details.
+#
+# You should have received a copy of the GNU General Public License
+# along with Ansible.  If not, see <http://www.gnu.org/licenses/>.
+#
+ANSIBLE_METADATA = {'status': ['preview'],
+                    'supported_by': 'community',
+                    'version': '1.0'}
+
+DOCUMENTATION = '''
+
+module: na_cdot_volume
+
+short_description: Manage NetApp cDOT volumes
+extends_documentation_fragment:
+    - netapp.ontap
+version_added: '2.3'
+author: Sumit Kumar (sumit4@netapp.com)
+
+description:
+- Create or destroy volumes on NetApp cDOT
+
+options:
+
+  state:
+    description:
+    - Whether the specified volume should exist or not.
+    required: true
+    choices: ['present', 'absent']
+
+  name:
+    description:
+    - The name of the lun to manage.
+    required: true
+
+  is_infinite:
+    description:
+    - Set True if the volume is an Infinite Volume.
+    required: false
+    choices: ['True', 'False']
+    default: 'False'
+
+  is_online:
+    description:
+    - Whether the specified volume is online, or not.
+    required: false
+    choices: ['True', 'False']
+    default: 'True'
+
+  aggregate_name:
+    description:
+    - The name of the aggregate the flexvol should exist on.
+    required: false
+    note: required when C(state=present)
+
+  size:
+    description:
+    - The size of the volume in (size_unit).
+    required: false
+    note: required when C(state=present)
+
+  size_unit:
+    description:
+    - The unit used to interpret the size parameter.
+    required: false
+    choices: ['bytes', 'b', 'kb', 'mb', 'gb', 'tb', 'pb', 'eb', 'zb', 'yb']
+    default: 'gb'
+
+  vserver:
+    description:
+    - Name of the vserver to use.
+    required: true
+    default: None
+
+'''
+
+EXAMPLES = """
+
+    - name: Create FlexVol
+      na_cdot_volume:
+        state: present
+        name: ansibleVolume
+        is_infinite: False
+        aggregate_name: aggr1
+        size: 20
+        size_unit: mb
+        vserver: ansibleVServer
+        hostname: "{{ netapp_hostname }}"
+        username: "{{ netapp_username }}"
+        password: "{{ netapp_password }}"
+
+    - name: Make FlexVol offline
+      na_cdot_volume:
+        state: present
+        name: ansibleVolume
+        is_infinite: False
+        is_online: False
+        vserver: ansibleVServer
+        hostname: "{{ netapp_hostname }}"
+        username: "{{ netapp_username }}"
+        password: "{{ netapp_password }}"
+
+"""
+
+RETURN = """
+
+
+"""
+
+"""
+    TODO:
+        Add more configurable parameters
+"""
+
+from ansible.module_utils.basic import AnsibleModule
+from ansible.module_utils.pycompat24 import get_exception
+import ansible.module_utils.netapp as netapp_utils
+
+HAS_NETAPP_LIB = netapp_utils.has_netapp_lib()
+
+
+class NetAppCDOTVolume(object):
+
+    def __init__(self):
+
+        self._size_unit_map = dict(
```

Comment:
I added one here any way because I realized there's a slight difference between how our different platforms report these sizes internally. Having them explicitly stated here allows easy maintenance/changes to these values in the future.